noir-projects/aztec-nr/authwit/src/account.nr

```diff
@@ -16,48 +19,49 @@ impl<Context> AccountActions<Context> {
 
 /**
  * An implementation of the Account Action struct for the private context.
- * 
+ *
  * Implements logic to verify authorization and execute payloads.
  */
 impl AccountActions<&mut PrivateContext> {
 
-    /** 
+    /**
      * Verifies that the `app_hash` and `fee_hash` are authorized and then executes them.
-     * 
+     *
      * Executes the `fee_payload` and `app_payload` in sequence.
      * Will execute the `fee_payload` as part of the setup, and then enter the app phase.
-     * 
+     *
      * @param app_payload The payload that contains the calls to be executed in the app phase.
      * @param fee_payload The payload that contains the calls to be executed in the setup phase.
      */
     // docs:start:entrypoint
     pub fn entrypoint(self, app_payload: AppPayload, fee_payload: FeePayload) {
         let valid_fn = self.is_valid_impl;
 
-        let fee_hash = fee_payload.hash();
-        assert(valid_fn(self.context, fee_hash));
+        let combined_payload_hash = poseidon2_hash_with_separator(
```

Comment:
i decided to go with the combined payload hash because it seems good enough and it required much less time than somehow merging `AppPayload` and `FeePayload` into 1 struct. Given that this is not enshrined in protocol I think we don't need to chase perfection here (which would require monster refactor).

Reply:
> I think we could optimize this a bit more and just add the app payload and fee payload hashes

Thinking about this more this could theoretically make it possible to swap the order in which fee payload and app payload are executed so it doesn't seem like a good idea.